app/presenters/tree_builder_vms_filter.rb

```diff
@@ -1,6 +1,6 @@
 class TreeBuilderVmsFilter < TreeBuilder
   def initialize(*args)
-    @root_class = 'ManageIQ::Providers::InfraManager::Vm'
+    @root_class ||= 'ManageIQ::Providers::InfraManager::Vm'
```

Comment:
I don't see how this can be previously set :disappointed:

Reply:
TreeBuilderImagesFilter, TreeBuilderVmsInstancesFilter, TreeBuilderInstancesFilter, TreeBuilderTemplatesImagesFilter and TreeBuilderTemplateFilter all set `@root_class` before calling `super` in the constructor.

Reply:
@skateman do you want this changed, or are you ok with this as is?

(We could probably just move the assignment *after* `super` in all the classes, solving the problem too.)

Reply:
oh, yes constructor redefinition call order, sure I'm fine